lightning/src/ln/channelmanager.rs

```diff
@@ -5633,7 +5599,8 @@ where
 									}
 
 									if let Some(ChannelPhase::Funded(ref mut chan)) = peer_state.channel_by_id.get_mut(&forward_chan_id) {
-										let (failure_code, data) = self.get_htlc_temp_fail_err_and_data(0x1000|7, short_chan_id, chan);
+										let failure_code = 0x1000|7;
+										let data = self.get_htlc_inbound_temp_fail_data(failure_code);
```

Comment:
IMHO there was no reason to change the function return type, is it?

Reply:
In case this function failed to fetch the `channel_update`, it previously indicated this to the caller by returning an `0x4000` error, and the caller would reassign the `0x1000` code with the `0x4000` code.

In case the `channel_update` fetch was successful, the returned error code was the same as the one passed in.

This function longer have this potential fetch failure, so it will never reassign the error code value to a different value. So we don't need to return an error code value any longer.

Reply:
Yeah I noted it, and make sense of your explaination. Probably just bias because the previous function was more "elegant" from the code point of view. Thanks for explaination